services/convert/user.go

```diff
@@ -37,16 +39,23 @@ func ToUsers(ctx context.Context, doer *user_model.User, users []*user_model.Use
 
 // ToUserWithAccessMode convert user_model.User to api.User
 // AccessMode is not none show add some more information
-func ToUserWithAccessMode(ctx context.Context, user *user_model.User, accessMode perm.AccessMode) *api.User {
+func ToUserWithAccessMode(ctx context.Context, user, owner *user_model.User, accessMode perm.AccessMode) *api.User {
 	if user == nil {
 		return nil
 	}
-	return toUser(ctx, user, accessMode != perm.AccessModeNone, false)
+	return toUser(ctx, user, owner, accessMode != perm.AccessModeNone, false)
 }
 
 // toUser convert user_model.User to api.User
 // signed shall only be set if requester is logged in. authed shall only be set if user is site admin or user himself
-func toUser(ctx context.Context, user *user_model.User, signed, authed bool) *api.User {
+func toUser(ctx context.Context, user, doer *user_model.User, signed, authed bool) *api.User {
+	starCount, _ := repo_model.CountRepository(ctx, &repo_model.SearchRepoOptions{
+		Actor:       doer,
+		Private:     signed,
+		StarredByID: user.ID,
+		Collaborate: util.OptionalBoolFalse,
```

Comment:
```suggestion
		Collaborate: optional.Some(false),
```

And import `code.gitea.io/gitea/modules/optional` above.